crates/sui-core/src/unit_tests/transaction_tests.rs

```diff
@@ -347,81 +388,94 @@ pub fn init_move_call_transaction(
 }
 
 async fn do_transaction_test_skip_cert_checks(
-    expected_sig_errors: u64,
     pre_sign_mutations: impl Fn(&mut TransactionData),
     post_sign_mutations: impl Fn(&mut Transaction),
     err_check: impl Fn(&SuiError),
 ) {
+    let (sender1, sender_key1): (_, AccountKeyPair) = get_key_pair();
+    let (sender2, sender_key2): (_, AccountKeyPair) = get_key_pair();
     do_transaction_test_impl(
-        expected_sig_errors,
         false,
+        &[(sender1, sender_key1), (sender2, sender_key2)],
+        Object::with_id_owner_for_testing(ObjectID::random(), sender1),
         pre_sign_mutations,
         post_sign_mutations,
+        0,
+        1,
         err_check,
     )
     .await
 }
 
 async fn do_transaction_test(
-    expected_sig_errors: u64,
     pre_sign_mutations: impl Fn(&mut TransactionData),
     post_sign_mutations: impl Fn(&mut Transaction),
     err_check: impl Fn(&SuiError),
 ) {
+    let (sender1, sender_key1): (_, AccountKeyPair) = get_key_pair();
+    let (sender2, sender_key2): (_, AccountKeyPair) = get_key_pair();
     do_transaction_test_impl(
-        expected_sig_errors,
         true,
+        &[(sender1, sender_key1), (sender2, sender_key2)],
+        Object::with_id_owner_for_testing(ObjectID::random(), sender1),
         pre_sign_mutations,
         post_sign_mutations,
+        0,
+        1,
         err_check,
     )
     .await
 }
 
 async fn do_transaction_test_impl(
-    _expected_sig_errors: u64,
     check_forged_cert: bool,
+    senders: &[(SuiAddress, AccountKeyPair)],
+    input_object: Object,
     pre_sign_mutations: impl Fn(&mut TransactionData),
     post_sign_mutations: impl Fn(&mut Transaction),
+    transfer_sender: usize,
+    move_call_sender: usize,
```

Comment:
it seems like it would be more readable to have 

```
transfer_sender: (SuiAddress, AccountKeyPair),
move_call_sender: (SuiAddress, AccountKeyPair),
```

instead of passing in a slice and indices into the slice separately

Reply:
yeah maybe this is slightly overkill, my thinking was you could potentially add more senders than just those two, for use as needed in the pre and post sign mutations. granted nothing is doing that yet.

Reply:
Acutally, duh, the new thing I wrote is already using it because it deliberately uses the wrong sender in the test